### PR TITLE
Validate Recipes Don't Contain Sub-recipes of Themselves

### PR DESCRIPTION
**Pull request by emimaesmith**
##### Checklist

- [x] `manage.py test` passes
- [x] tests are included

### Affected app(s)

recipe
### Description of change

Updated the Edit/Validate logic to check for sub-recipes of the same time of the recipe being edited. This validation will drill down into each sub-recipe to ensure we don't have a loop anywhere.
Addresses #1808

### CODE REVIEW

File: scale/recipe/models.py [hunk 1]
```diff
@@ -1281,6 +1281,16 @@ def edit_recipe_type_v6(self, recipe_type_id, title=None, description=None, defi
                 recipe_type.definition = convert_recipe_definition_to_v6_json(definition).get_dict()
             else:
                 raise InvalidDefinition('INVALID_DEFINITION', 'This version of the recipe definition is invalid to save')
+
+            # try:
```
Comment (gisjedi) [inline scale/recipe/models.py]: Meant to leave this commented out? If so, lets get rid of it.

File: scale/recipe/models.py [hunk 2]
```diff
@@ -1442,6 +1451,37 @@ def validate_recipe_type_v6(self, name, definition_dict):
 
         return RecipeTypeValidation(is_valid, errors, warnings, diff)
 
+    def validate_recursive_subs(self, recipe_type_name, definition, source_recipe_type_name=None):
+        """Validates the sub-recipes to ensure we don'' end up in a recipeception loop
+        :param recipe_type_name: The name of the recipe type we're looking for
+        :type recipe_type_name: String
+        :param definition: The RecipeDefinition we're searching
+        :type definition: RecipeDefinition
```
Comment (gisjedi) [inline scale/recipe/models.py]: Should do a full qualified python import so you can click through to definition in IDEs. Like so: https://github.com/ngageoint/scale/blob/master/scale/job/seed/manifest.py#L91

File: scale/recipe/models.py [hunk 3]
```diff
@@ -1442,6 +1451,37 @@ def validate_recipe_type_v6(self, name, definition_dict):
 
         return RecipeTypeValidation(is_valid, errors, warnings, diff)
 
+    def validate_recursive_subs(self, recipe_type_name, definition, source_recipe_type_name=None):
+        """Validates the sub-recipes to ensure we don'' end up in a recipeception loop
```
Comment (gisjedi) [inline scale/recipe/models.py]: don'' -> don't

File: scale/recipe/models.py (hunk 2 above)
Comment (emimaesmith) [reply scale/recipe/models.py]: fixed

**Comment by emimaesmith** [comment]
@gisjedi Ready for re-review

**Pull request merged**